### DISCUSSION TITLE
How to use template/double quotes to avoid escaping with "quotes" rule?

### DISCUSSION THREAD

**Discussion by ackmanx** [eslint/eslint#18031] [Q&A]
Given this config and the following strings:

```
/*eslint quotes: ["error", "single", { "avoidEscape": true }]*/
```

```
var single = `i'm so happy to use these quotes`;
var single2 = "i'm so happy to use these quotes";
```

The first one with the template literal fails validation, and ESLint "corrects" it to using a single quote _with escaping_

```var single = 'i\'m so happy to use these quotes';```

How do I allow using template literals to avoid escaping? It seems to work correctly if I use double quotes. Adding `"allowTemplateLiterals": true` doesn't work because then that also allows template literals without variables within them.

ESLint Playground:

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgcXVvdGVzOiBbXCJlcnJvclwiLCBcInNpbmdsZVwiLCB7IFwiYXZvaWRFc2NhcGVcIjogdHJ1ZSB9XSovXG5cbnZhciBzaW5nbGUgPSAnaVxcJ20gc28gaGFwcHkgdG8gdXNlIHRoZXNlIHF1b3Rlcyc7Iiwib3B0aW9ucyI6eyJlbnYiOnt9LCJydWxlcyI6e30sInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fSwiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIn19fQ==

**Comment by Rec0iL99** [comment]
Hi @ackmanx, could you be clearer as to what you are looking for?

Are you looking for a way to make the `quotes` rule not flag the template literal?

```js
/*eslint quotes: ["error", "single", { "avoidEscape": true }]*/

var single = `i'm so happy to use these quotes`;
```
